### DISCUSSION TITLE
Any way to add an initial loading screen

### DISCUSSION THREAD

**Discussion by manishprivet** [vercel/next.js#14495] [Help]
I am currently building a NextJS site for a client, which is quite resource heavy due to styling and interactions. I don't want to show the page to the user before the scripts arrive. 

Is there any way to add an initial loading screen which waits till the scripts arrive and then disappears?

In CRA, I was able to do this by adding the required HTML markup for loading screen in the index.html file, and then when scripts arrived, React took over and that markup disappeared.

[Edit] I was thinking that if there was a way to defer the next scripts, so that they don't render block the html, and then when the app mounts, set the display property to none or completely remove the loading div from html by javascript.

**Comment by IamLizu** [comment]
> I was thinking that if there was a way to defer the next scripts, so that they don't render block the html

Actually, what nextjs does is, when  your request a website it loads the HTML first and then it beatifies things with js. This is one of the major differences between react and nextjs. 

I would suggest you to use if possible, [react-loading-skeleton](https://www.npmjs.com/package/react-loading-skeleton)

**Reply by titungdup** [reply]
> Actually, what nextjs does is, when your request a website it loads the HTML first and then it beatifies things with js. This is one of the major differences between react and nextjs.

Does this mean we cannot implement initial loader in NextJS? @IamLizu

**Comment by eddsaura** [comment]
Hey, have you found some way to do this? We need to create an initial loader too

**Reply by manishprivet** [reply]
It is possible by a hack basically, e.g., make each page as a component, and then lazy load in their respective files in the pages folder, but I won't recommend it. Instead, we can lazy load all the components themselves, it doesn't affect performance as much, as Nextjs is performant in SSR.

**Comment by mike4ai** [comment]
skeleton is better than loading component, why?
